sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_message_backcompat.py

```diff
@@ -48,45 +72,49 @@ class MessageAlreadySettled(Exception):
 
 
 class LegacyMessage(object):  # pylint: disable=too-many-instance-attributes
-    def __init__(self, message, **kwargs):
+    def __init__(
+        self,
+        message: "AmqpAnnotatedMessage",
+        **kwargs: Any
+    ) -> None:
         self._message: "AmqpAnnotatedMessage" = message
-        self.state = MessageState.SendComplete
-        self.idle_time = 0
-        self.retries = 0
-        self._settler = kwargs.get("settler")
+        self.state: "MessageState" = MessageState.SendComplete
+        self.idle_time: int = 0
+        self.retries: int = 0
+        self._settler: "Settler" = kwargs.pop("settler")
         self._encoding = kwargs.get("encoding")
-        self.delivery_no = kwargs.get("delivery_no")
-        self.delivery_tag = kwargs.get("delivery_tag") or None
-        self.on_send_complete = None
-        self.properties = (
+        self.delivery_no: Optional[int] = kwargs.get("delivery_no")
+        self.delivery_tag: Optional[str] = kwargs.get("delivery_tag") or None
+        self.on_send_complete: Optional[Callable] = None
+        self.properties: Optional[LegacyMessageProperties] = (
             LegacyMessageProperties(self._message.properties)
             if self._message.properties
             else None
         )
-        self.application_properties = (
+        self.application_properties: Optional[Dict[Union[str, bytes], Any]] = (
             self._message.application_properties
             if self._message.application_properties and any(self._message.application_properties)
             else None
         )
-        self.annotations = (
+        self.annotations: Optional[Dict[Union[str, bytes], Any]] = (
             self._message.annotations
             if self._message.annotations and any(self._message.annotations)
             else None
         )
-        self.header = (
+        self.header: Optional[LegacyMessageHeader] = (
             LegacyMessageHeader(self._message.header) if self._message.header else None
         )
-        self.footer = self._message.footer
-        self.delivery_annotations = self._message.delivery_annotations
+        self.footer: Optional[Dict[Any, Any]]  = self._message.footer
+        self.delivery_annotations: Optional[Dict[Union[str, bytes], Any]] = self._message.delivery_annotations
         if self._settler:
             self.state = MessageState.ReceivedUnsettled
         elif self.delivery_no:
             self.state = MessageState.ReceivedSettled
-        self._to_outgoing_amqp_message: Callable = kwargs.get(
+        self._to_outgoing_amqp_message: Callable = cast(Callable, kwargs.get(
```

Comment:
we can type out this kwarg and not need the cast

Reply:
I can also pop this for now, then update for expanded kwargs in a separate PR.